src/model/element/interaction.js

```diff
@@ -12,6 +12,11 @@ const TYPE = 'interaction';
 const makeAssociation = ( type, intn ) => {
   let Type = _.isString(type) ? getIntnTypeByVal(type) : type;
 
+  // fall back on default type if garbage data is passed as `type` arg
+  if( Type == null || type == null ){
```

Comment:
Changes looks fine to me but I only stuck around this line. Is not ``Type`` already suppose to be assigned to ``null`` if ``type`` is ``null``?

Reply:
Strictly speaking, the extra check on `Type` is not necessary.  I thought it might be useful in case `getIntnTypeByVal()` ever changes -- as it's in another file -- or if we later wanted to use reflection at the end of the `if` to validate the class of `Type`.

Basically, my goal was to make the checks as conservative as possible to help prevent future problems when the code changes.

Reply:
Actually, what I exactly mean was that ``_.isString(type)`` returns ``false`` if type is ``null`` and ``Type`` is assigned to ``null`` in that case. Therefore, ``Type == null || type == null`` must be equal to ``Type == null`` for the current status of the code. However, I think you would not want to change the code if how ``Type`` is calculated changes in a way that the rule I mentioned gets lost. 

That makes sense, I just wanted to make sure that you did it on purpose.